prompting-client/src/snapd_client/prompt.rs

```diff
@@ -205,6 +205,22 @@ impl TypedUiInput {
 #[derive(Debug, Default, Clone, PartialEq, Eq, Deserialize, Serialize)]
 pub struct PromptId(pub String);
 
+#[derive(Debug, Clone, PartialEq, Eq, Deserialize, Serialize)]
+pub enum PromptNotice {
+    Update(PromptId),
+    Resolved(PromptId),
+}
+
+impl PromptNotice {
+    /// Flatten this notice into the enclosed ID if it was an update
+    pub fn into_option_id(self) -> Option<PromptId> {
```

Comment:
Is that a typical naming convention or should this be called `into_prompt_id`?

Reply:
`into_foo` is the standard naming convention for a method taking `self` and returning a `Foo`, so arguably it should be `into_option_prompt_id` but that felt a little verbose. (`into_prompt_id` would be `self -> PromptId`) There is also `try_into_foo` but for the most part that is something that would return a `Result`.

There are some nice guidelines [here](https://rust-lang.github.io/api-guidelines/naming.html) for a bunch of API related things relating to Rust crates.

The issue is that this is only needed for the integration tests so we either need a public method or we need to make the fields inside of the `PromptNotice` enum public. I tend to avoid the latter as it prevents users of the type outside of the module from messing around with the internals or constructing the type but maybe that would be better in this case?